clang/lib/AST/ItaniumMangle.cpp

```diff
@@ -5828,6 +5822,15 @@ void CXXNameMangler::mangleExpression(const Expr *E, unsigned Arity,
     break;
   }
 
+  case Expr::PackIndexingExprClass: {
+    auto *PE = cast<PackIndexingExpr>(E);
+    NotPrimaryExpr();
```

Comment:
I am looking at the test cases and I don't see the `X`, can we make sure we cover that branch in `NotPrimaryExpr` e.g. `if (AsTemplateArg && IsPrimaryExpr)`. We should make sure we cover that in the demangling as well.